src/theory/arith/linear/theory_arith_private.cpp

```diff
@@ -3439,20 +3439,53 @@ bool TheoryArithPrivate::postCheck(Theory::Effort effortLevel)
     }
 
     if(!emmittedConflictOrSplit) {
-      std::vector<TrustNode> possibleLemmas = roundRobinBranch();
-      if (!possibleLemmas.empty())
+      bool tryNew;
+      std::unordered_set<Node> lemmas;
+      Trace("arith-round-robin") << "Round robin branch..." << std::endl;
+      do
       {
-        ++(d_statistics.d_externalBranchAndBounds);
-        d_cutCount = d_cutCount + 1;
-        for (const TrustNode& possibleLemma : possibleLemmas)
+        tryNew = false;
+        std::vector<TrustNode> possibleLemmas = roundRobinBranch();
+        Trace("arith-round-robin") << "...consider " << possibleLemmas.size()
+                                   << " lemmas" << std::endl;
+        if (!possibleLemmas.empty())
         {
-          Trace("arith::lemma") << "rrbranch lemma" << possibleLemma << endl;
-          if (outputTrustedLemma(possibleLemma, InferenceId::ARITH_BB_LEMMA))
+          ++(d_statistics.d_externalBranchAndBounds);
+          d_cutCount = d_cutCount + 1;
+          for (const TrustNode& possibleLemma : possibleLemmas)
           {
-            emmittedConflictOrSplit = true;
+            Node lem = possibleLemma.getProven();
+            if (lemmas.find(lem) != lemmas.end())
+            {
+              Trace("arith-round-robin") << "..already fail lemma" << std::endl;
+              continue;
+            }
+            lemmas.insert(lem);
```

Comment:
Move to if branch to avoid potential false flags (this would only happen if `possibleLemmas` contains duplicate lemmas in the same while loop iteration).

Reply:
The termination argument is based on adding it here. I will comment.